generate/src/contract/events.rs

```diff
@@ -247,13 +247,138 @@ fn expand_all_events(cx: &Context) -> TokenStream {
         return quote! {};
     }
 
+    let event_enum = expand_event_enum(cx);
+    let event_from_log = expand_event_from_log(cx);
+
     quote! {
         impl Contract {
             /// Returns a log stream with all events.
-            pub fn all_events(&self) -> self::ethcontract::dyns::DynAllEventsBuilder<
-                self::ethcontract::RawEventData,
-            > {
-                self.raw_instance().all_events()
+            pub fn all_events(&self) -> self::ethcontract::dyns::DynAllEventsBuilder<Event> {
+                self::ethcontract::dyns::DynAllEventsBuilder::new(
+                    self.raw_instance().web3(),
+                    self.address(),
+                )
+            }
+        }
+
+        #event_enum
+        #event_from_log
+    }
+}
+
+/// Expands into an enum with one variant for each distinct event type,
+/// including anonymous types.
+fn expand_event_enum(cx: &Context) -> TokenStream {
+    let variants = cx
+        .artifact
+        .abi
+        .events()
+        .map(|event| {
+            let struct_name = expand_struct_name(&event);
+            quote! {
+                #struct_name(self::events::#struct_name)
+            }
+        })
+        .collect::<Vec<_>>();
+
+    quote! {
+        /// A contract event.
+        pub enum Event {
+            #( #variants, )*
```

Comment:
I think this creates `enum Event { Deposit, Withdraw }` but we want `enum Event { Deposit(events::Deposit), Withdraw(events::Withdraw)  }`

Reply:
Each variant is `#struct_name(self::events::#struct_name)`.